nix/nixos-module.nix

```diff
@@ -161,29 +170,52 @@ in
     };
   };
 
-  config = lib.mkIf cfg.enable {
-    environment.systemPackages = [ cfg.package ];
+  config = lib.mkMerge [
+    (lib.mkIf cfg.enable {
+      environment.systemPackages = [ cfg.package ];
 
-    users.groups.uinput = { };
-    users.groups.kmonad = { };
+      users.groups.uinput = { };
+      users.groups.kmonad = { };
 
-    users.users.kmonad = {
-      description = "KMonad system user.";
-      group = "kmonad";
-      isSystemUser = true;
-    };
+      users.users.kmonad = {
+        description = "KMonad system user";
+        group = "kmonad";
+        isSystemUser = true;
+      };
 
-    services.udev.extraRules = ''
-      # KMonad user access to /dev/uinput
-      KERNEL=="uinput", MODE="0660", GROUP="uinput", OPTIONS+="static_node=uinput"
-    '';
```

Comment:
This is a bit mangled in these changes; I think we should perhaps split it out into a separate one?